### ISSUE TITLE
Possible endless loop in modula lexer

### ISSUE THREAD

**Issue by stefankueng** [ScintillaOrg/lexilla#128]
https://github.com/ScintillaOrg/lexilla/blob/557d750299f2c27b0f35e219e686602eb5f6ff43/lexers/LexModula.cxx#L269

this loop might never end:
```
while( ( ch = styler.SafeGetCharAt( pos ) ) != '\n' ) {
    if( ch == 'P' ) {
        if( styler.StyleAt(pos) == SCE_MODULA_KEYWORD ) {
            if( checkKeyIdentOper( styler, pos, endPos,
                                "PROCEDURE", '(' ) ) {
                break;
            }
        }
    }
    pos++;
}
```

`pos `gets incremented endlessly if a char `'P'` is never found in the last line of a document. The while loop should check for end-of-document as well.

**Comment by nyamatongwe** [comment]
Please attach an example file that illustrates the problem.

**Comment by stefankueng** [comment]
[Form.zip](https://github.com/ScintillaOrg/lexilla/files/10471895/Form.zip)

open this file in scite, then select the modula-3 language for it (you may have to enable modula-3 first in the global config file). 
then open another file (doesn't matter what).
switch between the two tabs. maybe scroll to the end of the form.cp file to get the effect.

I know that form.cp is not a modula file, but even though the wrong lexer is used, it should never hang.

**Comment by nyamatongwe** [comment]
Reduced the example to 15 lines to be more tractable although the trigger still isn't clear.
~~~modula-3
MODULE Form;
  IMPORT  

  PROCEDURE (bf: ButtonForm) InitializeComponent(), NEW;
  BEGIN
    bf.SuspendLayout();
    REGISTER(bf.button1.Click, bf.button1_Click);
    bf.get_Controls().Add(bf.button2);
  END InitializeComponent;

BEGIN
    NEW(bf);
    Wfm.Application.Run(bf);
END Form.
~~~
[F1.cp.txt](https://github.com/ScintillaOrg/lexilla/files/10475540/F1.cp.txt)

**Comment by nyamatongwe** [comment]
Fixed with 302e75a788870d91ccade8172e9eadb0c3829210.

**Comment by nyamatongwe** [comment]
Fixed in 5.2.2.
